### ISSUE TITLE
Shadowing bug in inlining affecting foreach

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/ergo#639]
**Describe the bug**
Inlining does not handle local variable shadowing properly when within `foreach` expressions.

**To Reproduce**

The following zip contains a simple test:
[inlinetest.zip](https://github.com/accordproject/ergo/files/3433680/inlinetest.zip)

Unzip the file and execute the contract with the following command:
```
ergorun.js execute model.cto logic.ergo --contract contract.json --request request.json  
```

For the following expression in the test clause:
```
    return MyResponse{
			output1: let c = 2; c,
			output2: foreach c in [1,2,3] return c
		}
```

The result is:
```
{
    "$class": "org.accordproject.inline.MyResponse",
    "output1": 2,
    "output2": [1, 1, 1],
    "transactionId": "07c93b3c-2750-46aa-a8a7-488cf9ade69c",
    "timestamp": "2019-07-26T00:13:41.028Z"
  }
```

The local variable `c` shadows the global constant `c` properly for the `let` expression, but not for the `foreach` expression.

**Expected behavior**

The expected result should be:
```
{
    "$class": "org.accordproject.inline.MyResponse",
    "output1": 2,
    "output2": [1, 2, 3],
    "transactionId": "07c93b3c-2750-46aa-a8a7-488cf9ade69c",
    "timestamp": "2019-07-26T00:13:41.028Z"
  }
```

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser N/A
 - Version 0.9

**Comment by jeromesimeon** [comment]
The most likely reason for the bug is that the generic mapper for expressions used inside the inlining phase, does not record variables from the foreach expression properly in the context:

The corresponding code is at:
https://github.com/accordproject/ergo/blob/6e755cd9a246f179c056806a1f961534b0e24503/mechanization/Ergo/Lang/ErgoMap.v#L102

Which does not call the `ctxt_new_variable_scope`.

In contrast, here is the corresponding call for `let`:
https://github.com/accordproject/ergo/blob/6e755cd9a246f179c056806a1f961534b0e24503/mechanization/Ergo/Lang/ErgoMap.v#L73

**Comment by jeromesimeon** [comment]
Addressed in #640
